osquery/tables/system/posix/ssl_cert.cpp

```diff
@@ -0,0 +1,199 @@
+/*
+ *  Copyright (c) 2014-present, Facebook, Inc.
+ *  All rights reserved.
+ *
+ *  This source code is licensed under the BSD-style license found in the
+ *  LICENSE file in the root directory of this source tree. An additional grant
+ *  of patent rights can be found in the PATENTS file in the same directory.
+ *
+ */
+
+#include <stdio.h>
+#include <string.h>
+#include <string>
+
+#include <osquery/tables.h>
+
+#include <openssl/asn1.h>
+#include <openssl/bio.h>
+#include <openssl/bn.h>
+#include <openssl/err.h>
+#include <openssl/pem.h>
+#include <openssl/x509.h>
+
+#include <boost/filesystem/operations.hpp>
+#include <boost/filesystem/path.hpp>
+
+#define SHORT_STR 256
+
+namespace osquery {
+namespace tables {
+
+namespace fs = boost::filesystem;
+
+static void fillRow(Row& r, X509* cert) {
+  char temp[SHORT_STR + 1];
+
+  // set certificate subject information
+  X509_NAME* subject_name = X509_get_subject_name(cert);
```

Comment:
You might be able to get away with `auto subject_name = X509_get_subject_name(cert);` here?